File: leetcode/559-maximum-depth-of-n-ary-tree/main.go
package main

type NaryNode struct {
	Val      int
	Children []*NaryNode
}

type Stack struct {
	Node  *NaryNode
	Depth int
}

func maxDepth(root *NaryNode) int {
	if root == nil {
		return 0
	}
	result := 0
	var stack []*Stack
	stack = append(stack, &Stack{root, 0})
	for len(stack) > 0 {
		pop := stack[len(stack)-1]
		stack = stack[:len(stack)-1]
		if pop.Depth > result {
			result = pop.Depth
		}
		for i := 0; i < len(pop.Node.Children); i++ {
			stack = append(stack, &Stack{pop.Node, pop.Depth + 1})
		}
	}
	return result
}

func main() {

}

File: leetcode/559-maximum-depth-of-n-ary-tree/main.py
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Node
        :rtype: int
        """
        if root is None:
            return 0
        result = 0
        stack = []
        stack.append((root, 1))
        while len(stack) > 0:
            pop = stack.pop()
            if pop[1] > result:
                result = pop[1]
            for i in range(len(pop[0].children)):
                stack.append((pop[0].children[i], pop[1]+1))
        return result